magento/pwa-studio
# [PWA-1709] Consolidate and use WishlistButton

## Description

As a user I want a consistent UX with regards to wishlists.

As a developer I want a consistent component / coding experience.

Preconditions

Wishlist is enabled in Magento Admin Panel > Stores > Settings > Configuration > Customers > Wish List
Multi-wishlist is not enabled in Stores > Settings > Configuration > Customers > Wish List > Enable Multiple Wish Lists (EE) OR Maximum number of wishlists is set to one (1)
Acceptance Criteria

Login toasts are rendered if user clicks to add to wishlist but is not signed in
If EE only one wishlist is allowed, no dialog (to choose a wishlist) should appear
The "general" Wishlist Button component takes props such as localized "buttonText" string for customization
The "general" Wishlist Button component takes props for beforeAdd and afterAdd callbacks

The goal of this story is to replace any other UX/logic with the existing WishlistButton component, with some changes such as:

- Rendering a login toast if user clicks to add but is not signed in.
- Modify EE functionality to act as CE if only 1 wishlist is allowed (no dialog popup)
- Take props such as "buttonText" string
- Take props for beforeAdd and afterAdd callbacks

## Related Issue







* [[PWA-1709](https://jira.corp.magento.com/browse/PWA-1709)] Consolidate and use WishlistButton

## Acceptance





### Verification Stakeholders



### Specification




## Test Plan






#### Test scenario(s) for direct fix/feature
1. Navigate to a category page
2. Verify that the wishlist button is present on each product tile
3. Verify that when the user is not signed in, clicking the wishlist button pops a "you need to login first" toast
4. Verify that once the user is signed in, clicking the wishlist button adds the product to the single wishlist. No dialog to choose which wishlist appears (based on Preconditions).
5. Navigate to a product page
6. Verify that the wishlist button is always present (signed in and not)
7. Verify that when the user is not signed in, clicking the wishlist button pops a "you need to login first" toast
8. Verify that once the user is signed in, clicking the wishlist button adds the product to the single wishlist. No dialog to choose which wishlist appears ((based on Preconditions).
9. Add an item to Cart and navigate to /cart page
10. Verify that when the user is not signed in, clicking the wishlist button pops a "you need to login first" toast
11. Verify that once the user is signed in, clicking the wishlist button adds the product to the single wishlist. No dialog to choose which wishlist appears (based on Preconditions).

#### Test scenario(s) for any existing impacted features/areas
Example
-   Verify user is able to sort data after applying filters on category page
-   Verify user is able to sort data after applying filters on search page

#### Test scenario(s) for any Magento Backend Supported Configurations
Example
- Update default Sort value in backend and repeat above scenarios

#### Is Browser/Device testing needed?
Example
- Yes, browser testing is needed as X UI component may be impacted on <browser>
- Yes, device testing is needed as X UI component may be impacted on <mobile|desktop|etc>

#### Any ad-hoc/edge case scenarios that need to be considered?
Example
- Apply all filters to get 0 results and then remove filters to see respective products 

## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
